components/cli/pkg/commands/instance/logs.go

```diff
@@ -24,32 +24,32 @@ import (
 	"github.com/cellery-io/sdk/components/cli/cli"
 )
 
-func RunLogs(cli cli.Cli, cellName, componentName string, sysLog bool) error {
+func RunLogs(cli cli.Cli, instanceName string, componentName string, sysLog bool, follow bool) error {
 	if componentName == "" {
-		var logs string
+		var instanceAvailable bool
 		var err error
 		if sysLog {
-			logs, err = cli.KubeCli().GetCellLogsAllComponents(cellName)
+			instanceAvailable, err = cli.KubeCli().GetCellLogsAllComponents(instanceName, follow)
```

Comment:
Shall we update the logic in the "RunLogs" function? We can use the newly implemented "GetInstanceAvailability" function to check the instance availability. 

IMHO it is better if we could remove the responsibility of checking the instance availability from "GetCellLogsUserComponents", "GetComponentLogs" functions. They should only return the logs or errors. It is better if this function could be updated as follows.

1. Check instance availability at the beginning - return "instance not found" error if not available.
2. with "--follow" flag - only return if there is an error.
3. without "--follow" flag - print "No logs found for instance <NAME>" and not return an error. Since this is not an error.
WDYT?

Reply:
+1